earthfile2llb/wait_block.go

```diff
@@ -0,0 +1,187 @@
+package earthfile2llb
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/earthly/earthly/states"
+	"github.com/earthly/earthly/util/gatewaycrafter"
+	"github.com/earthly/earthly/util/llbutil"
+	"github.com/earthly/earthly/util/llbutil/pllb"
+	"github.com/earthly/earthly/util/syncutil/serrgroup"
+
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/pkg/errors"
+)
+
+type saveImageWaitItem struct {
+	c    *Converter
+	si   states.SaveImage
+	push bool
+}
+
+type runCommandWaitItem struct {
+	c   *Converter
+	cmd *pllb.State
+}
```

Comment:
I wonder if this could be simplified even further - rather than add every run command to the wait block, what if we simply force execution of the current target's state at the end of every WAIT block?

Reply:
This has been changed to only wait during finalizeState, or when a new Build is triggered.